pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -156,7 +160,16 @@ func determineSeed(
 	if err != nil {
 		return nil, err
 	}
-	return getSeedWithLeastShootsDeployed(filteredSeeds, shootList.Items)
+
+	scoreFns := []scoreFn{
+		seedWithLeastShootsDeployedFn(shootList.Items),
+	}
+
+	if gardenerschedulerfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
+		scoreFns = append(scoreFns, seedWithPreferredHASetting(shoot))
```

Comment:
Scoring is applied partially. Irrespective of whether the configured HA type is `single-zone` or `multi-zone` or no-HA score should be computed. What we have now is if the value is `multi-zone` then we filter out the ones that are multi-zonal and then on top of that we also apply a scoring function via `seedWithPreferredHASetting` which have no value.